File: 프로그래머스/0/120863. 다항식 더하기/다항식 더하기.js
function solution(polynomial) {
    const arr = polynomial.replaceAll(" ", "").match(/([+/-]?\s*\d*\s*x)|([+/-]?\s*\d+)/g);
    let xNum = 0;
    let num = 0;
    
    arr.forEach((x) => {
        if(x.includes("x")) {
            if(x==="x" || x==="+x") xNum += 1;
            else if(x==="-x") xNum -= 1;
            else xNum += parseInt(x);
        } else {
            num += parseInt(x);
        }
    })
    return `${xNum===0 ? "" : (xNum===1 ? "x" : xNum+"x")}`+ `${num===0 ? "" : (xNum===0 ? ""+num : (num>0 ? " + "+num : num.toString().replace("-", " - ") ))}`;
}

File: 프로그래머스/0/120863. 다항식 더하기/다항식 더하기.py
def solution(polynomial):
    polynomial = polynomial.split()[::2]
    i, j = 0, 0

    for k in polynomial:
        if 'x' in k:
            if k == 'x':
                i += 1
            else:
                i += int(k[:-1])
        else:
            j += int(k)

    if i == 1 and j == 0:
        return "x"
    elif i == 1 and j > 0:
        return f"x + {j}"
    elif j == 0:
        return f"{i}x"
    elif i == 0:
        return f"{j}"
    else:
        return f"{i}x + {j}"